network/endpoint.go

```diff
@@ -128,14 +127,14 @@ func (nw *network) newEndpoint(
 	nl netlink.NetlinkInterface,
 	plc platform.ExecClient,
 	netioCli netio.NetIOInterface,
-	epInfo *EndpointInfo,
+	epInfo []*EndpointInfo,
 ) (*endpoint, error) {
 	var ep *endpoint
 	var err error
 
 	defer func() {
 		if err != nil {
-			log.Printf("[net] Failed to create endpoint %v, err:%v.", epInfo.Id, err)
+			log.Printf("[net] Failed to create endpoint %v, err:%v.", epInfo[0].Id, err)
```

Comment:
why are we hardcoding 0 index here? what if secondary endpoint failed.? the best way is to print error returned from internal function and that should return which endpoint failed

Reply:
this is passed from network.go though

Reply:
thats not what i meant.. since there are multiple endpoints now..return which endpoint failed from createendpoint api instead of hardcoding here